terraform/gcp/modules/monitoring/infra/alerts.tf

```diff
@@ -246,3 +246,89 @@ resource "google_monitoring_alert_policy" "kms_crypto_request_alert" {
   notification_channels = local.notification_channels
   project               = var.project_id
 }
+
+### Kubernetes Alerts
+
+# Kubernetes Container Memory Utilization > 90%
+resource "google_monitoring_alert_policy" "k8s_container_memory_utilization" {
+  # In the absence of data, incident will auto-close in 7 days
+  alert_strategy {
+    auto_close = "604800s"
+  }
+
+  combiner = "OR"
+
+  conditions {
+    condition_threshold {
+      aggregations {
+        alignment_period   = "300s"
+        per_series_aligner = "ALIGN_MEAN"
+      }
+
+      comparison      = "COMPARISON_GT"
+      duration        = "0s"
+      filter          = "metric.type=\"kubernetes.io/container/memory/request_utilization\" resource.type=\"k8s_container\""
```

Comment:
I think this presumes we have requests set on all containers, otherwise this will never fire? 

Would we also want an alert if the actual memory of the nodes exceeds some thresholds (regardless of requested allocations)?

Reply:
Updated to use the allocatable metric instead of request. We had also discussed potentially having a lower priority alert for limit utilization? I haven't been able to find any priority options for gcp alerts, do we currently have any low priority alerts? Do they just utilize different notification channels?